File: Names Scores/js/Names scores .js

// https://github.com/sarthakjain95/ProjectEulerSolutions
// Problem 22
// Names scores

// Please try attempting the problem yourself before
// looking at the solution.
// ...

let names_list= [];
const fs = require('fs');

try{  
	var data = fs.readFileSync('names.txt', 'utf8');
	// console.log(data.toString()); 
	names_list= data.toString().trim().split(',');
	// console.log(names_list);   
}catch(e){
	console.log('Error:', e.stack);
}

const add= (a,b)=>a+b;
const name_to_score= name=>name.replace(/\"/g,'').split('').map( ch=>(ch.charCodeAt(0)-64) ).reduce(add);

names_list= names_list.sort();
total_names_score= names_list.map( (e,i)=>name_to_score(e)*(i+1) ).reduce( add )

console.log( total_names_score );

// ProjectEulerSolutions
// Sarthak Jain

File: Names Scores/py/Names scores .py

# https://github.com/sarthakjain95/ProjectEulerSolutions
# Problem 22
# Names scores

# Please try attempting the problem yourself before
# looking at the solution.
# ...

names_file= open("names.txt",'r')
names_list= names_file.read()
names_file.close()

names_list= names_list.strip().split(',')
names_list.sort()

total_score= 0

for d in range(0,len( names_list)):

	name_score= 0
	
	for c in list(names_list[d].replace('"','')):
		name_score+= ord(c) - 64
	name_score*= d+1
	
	total_score+= name_score
	# print( names_list[d], name_score)


print("Total of all Name Scores is:",total_score)

# ProjectEulerSolutions
# Sarthak Jain